homeassistant/components/dormakaba_dkey/config_flow.py

```diff
@@ -0,0 +1,200 @@
+"""Config flow for Dormakaba dKey integration."""
+from __future__ import annotations
+
+from asyncio import Task
+import logging
+from typing import Any
+
+from bleak import BleakError
+from py_dormakaba_dkey import DKEYLock, device_filter, errors as dkey_errors
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.bluetooth import (
+    BluetoothServiceInfoBleak,
+    async_discovered_service_info,
+)
+from homeassistant.const import CONF_ADDRESS
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import ASSOCIATION_DATA, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_ASSOCIATE_SCHEMA = vol.Schema(
+    {
+        vol.Required("activation_code"): str,
+    }
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Dormakaba dKey."""
+
+    VERSION = 1
+    _lock: DKEYLock
+
+    def __init__(self) -> None:
+        """Initialize the config flow."""
+        self._connect_task: Task | None = None
+        # Populated by user step
+        self._discovered_devices: dict[str, BluetoothServiceInfoBleak] = {}
+        # Populated by bluetooth and user steps
+        self._discovery_info: BluetoothServiceInfoBleak | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the user step to pick discovered device."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            address = user_input[CONF_ADDRESS]
+            await self.async_set_unique_id(address)
+            self._abort_if_unique_id_configured()
+            self._discovery_info = self._discovered_devices[address]
+            return await self.async_step_bluetooth_connect()
+
+        current_addresses = self._async_current_ids()
+        for discovery in async_discovered_service_info(self.hass):
+            if (
+                discovery.address in current_addresses
+                or discovery.address in self._discovered_devices
+                or not device_filter(discovery.advertisement)
+            ):
+                continue
+            self._discovered_devices[discovery.address] = discovery
+
+        if not self._discovered_devices:
+            return self.async_abort(reason="no_unconfigured_devices")
```

Comment:
I recommend following our standard patterns for the flow, eg for strings. See eg the xiaomi_ble flow.

https://github.com/home-assistant/core/blob/ecb1d93b2e4455158a14ee8015625e5a0bc25cb9/homeassistant/components/xiaomi_ble/config_flow.py#L252-L253

Reply:
Hmm, I was using the `led_ble` integration as inspiration, this is indeed better 👍

Reply:
Thanks added to my list to adjust `led_ble` `yalexs_ble` `switchbot` `ld2410_ble` `keymitt_ble`